src/btree/bt_debug.c

```diff
@@ -994,17 +1004,20 @@ __wt_debug_tree(void *session_arg, WT_BTREE *btree, WT_REF *ref, const char *ofi
  *     Dump the in-memory information for a page.
  */
 int
-__wt_debug_page(
-  void *session_arg, WT_BTREE *btree, WT_REF *ref, const char *ofile, bool dump_app_data)
+__wt_debug_page(void *session_arg, WT_BTREE *btree, WT_REF *ref, const char *ofile,
+  bool dump_all_data, bool dump_key_data)
 {
     WT_DBG *ds, _ds;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
     uint32_t flags;
 
     flags = WT_DEBUG_TREE_LEAF;
-    if (dump_app_data)
-        LF_SET(WT_DEBUG_UNREDACT);
+    WT_ASSERT(NULL, !(dump_all_data && dump_key_data));
```

Comment:
Cast the session earlier so we can pass it here.

Reply:
fixed https://github.com/wiredtiger/wiredtiger/pull/10094/commits/96d06abcdc1a848fb2a55109f4789c72e69c24ea